navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -295,10 +327,26 @@ public void startNavigation(@NonNull DirectionsRoute directionsRoute) {
     this.directionsRoute = directionsRoute;
     Timber.d("MapboxNavigation startNavigation called.");
     if (!isBound) {
+      // Navigation sessions initially starting
+      sessionState = SessionState.builder()
+        .originalDirectionRoute(directionsRoute)
+        .currentDirectionRoute(directionsRoute)
+        .sessionIdentifier(TelemetryUtils.buildUUID())
+        .startTimestamp(new Date())
+        .rerouteCount(0)
+        .mockLocation(locationEngine instanceof MockLocationEngine)
+        .build();
+
       Intent intent = getServiceIntent();
       context.startService(intent);
       context.bindService(intent, this, Context.BIND_AUTO_CREATE);
       navigationEventDispatcher.onNavigationEvent(true);
+    } else {
+      // New directionRoute provided
+      sessionState = sessionState.toBuilder()
+        .currentDirectionRoute(directionsRoute)
+        .rerouteCount(sessionState.rerouteCount() + 1)
```

Comment:
iOS doesn't increment the rerouteCount if the developer manually set a new route:
https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxCoreNavigation/RouteController.swift#L142-L147

It only increments the reroute counter if the SDK performs an automatic reroute: https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxCoreNavigation/RouteController.swift#L631-L633
https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxCoreNavigation/RouteController.swift#L878

Reply:
SDK never automatically reroutes on Android? The user makes a new request for a route and passes it in here which, by definition, is a reroute.